internal/wire/op_msg.go

```diff
@@ -205,8 +205,6 @@ func (msg *OpMsg) readFrom(bufr *bufio.Reader) error {
 		if err := binary.Read(bufr, binary.LittleEndian, &msg.checksum); err != nil {
 			return lazyerrors.Error(err)
 		}
-
-		// TODO validate checksum https://github.com/FerretDB/FerretDB/issues/1626
```

Comment:
As I like the idea of doing checksum validation earlier, I believe that we'll be better if we move the validation code to this `readFrom` function. 
As this function initialize the message from reader, it also does lot of additional validation and returns the proper error if raw data from reader is not a correct message.

Also we already read the checksum and store it here, so we don't need to do it twice :)

Reply:
Yeah this would have been a lot easier. The issue with that is the header is missing. I need the header to calculate the checksum

Reply:
You're right, let's keep it where it is

Reply:
Cool